addons/product/models/product_catalog_mixin.py

```diff
@@ -86,23 +86,34 @@ def _get_product_catalog_order_line_info(self, product_ids, child_field=False, *
                 'quantity': float (optional)
                 'productType': string
                 'price': float
+                'uom': dict
                 'readOnly': bool (optional)
             }
         """
         order_line_info = {}
         default_data = self._default_order_line_values(child_field)
 
         for product, record_lines in self._get_product_catalog_record_lines(product_ids, child_field=child_field, **kwargs).items():
+            uom = {
+                'display_name': product.uom_id.display_name,
+                'id': product.uom_id.id,
+            }
             order_line_info[product.id] = {
                **record_lines._get_product_catalog_lines_data(parent_record=self, **kwargs),
                'productType': product.type,
+               'uom': uom,
             }
             product_ids.remove(product.id)
 
         products = self.env['product.product'].browse(product_ids)
         product_data = self._get_product_catalog_order_data(products, **kwargs)
         for product_id, data in product_data.items():
-            order_line_info[product_id] = {**default_data, **data}
+            product = self.env['product.product'].browse(product_id)
+            order_line_info[product_id] = {
+                **default_data,
+                **data,
+                'uom': {'display_name': product.uom_id.display_name, 'id': product.uom_id.id},
+            }
```

Comment:
Please don't browse the product (again) here. You can add `uom` directly form `_get_product_catalog_order_data` or simply add it in the kanban template to avoid sending it to javascript. 
```suggestion
            order_line_info[product_id] = {
                **default_data,
                **data,
            }
```